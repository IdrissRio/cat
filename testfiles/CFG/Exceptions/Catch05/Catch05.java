public class Catch05 {
  void test() {
    try {
      try {
        m();
      } catch (TestSubException e) {
        m();
      }
    } catch (TestException e) {
      int a;
    }
    int b;
  }

  void m() throws TestException, TestSubException {}

  class TestException extends RuntimeException {}
  class TestSubException extends TestException {}
}