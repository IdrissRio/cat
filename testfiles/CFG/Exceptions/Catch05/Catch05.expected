NodeName: Entry[15] Successors: [Exit[16]]
NodeName: Exit[16] Successors: []
NodeName: Entry[7] Successors: [Exit[8]]
NodeName: Exit[8] Successors: []
NodeName: Entry[29] Successors: [MethodAccess[45]]
NodeName: MethodAccess[45] Successors: [UncheckedExceptions[65], UncheckedExceptions[46]]
NodeName: UncheckedExceptions[65] Successors: [UncheckedExceptions[83], Exit[36]]
NodeName: UncheckedExceptions[83] Successors: [VariableDeclarator[89], Exit[36]]
NodeName: VariableDeclarator[89] Successors: [Exit[36]]
NodeName: Exit[36] Successors: []
NodeName: UncheckedExceptions[46] Successors: [ParameterDeclaration[52], ParameterDeclaration[68], Exit[36]]
NodeName: ParameterDeclaration[52] Successors: [MethodAccess[59]]
NodeName: MethodAccess[59] Successors: [UncheckedExceptions[83], UncheckedExceptions[60]]
NodeName: UncheckedExceptions[60] Successors: [ParameterDeclaration[68], Exit[36]]
NodeName: ParameterDeclaration[68] Successors: [VariableDeclarator[79]]
NodeName: VariableDeclarator[79] Successors: [VariableDeclarator[89]]
NodeName: Entry[93] Successors: [Exit[102]]
NodeName: Exit[102] Successors: []
NodeName: Entry[115] Successors: [Exit[116]]
NodeName: Exit[116] Successors: []
NodeName: Entry[107] Successors: [Exit[108]]
NodeName: Exit[108] Successors: []
NodeName: Entry[139] Successors: [Exit[140]]
NodeName: Exit[140] Successors: []
NodeName: Entry[131] Successors: [Exit[132]]
NodeName: Exit[132] Successors: []

NodeName: Exit[16] Predecessor: [Entry[15]]
NodeName: Entry[15] Predecessor: []
NodeName: Exit[8] Predecessor: [Entry[7]]
NodeName: Entry[7] Predecessor: []
NodeName: Exit[36] Predecessor: [UncheckedExceptions[83], UncheckedExceptions[65], UncheckedExceptions[46], UncheckedExceptions[60], VariableDeclarator[89]]
NodeName: UncheckedExceptions[83] Predecessor: [UncheckedExceptions[65], MethodAccess[59]]
NodeName: UncheckedExceptions[65] Predecessor: [MethodAccess[45]]
NodeName: MethodAccess[45] Predecessor: [Entry[29]]
NodeName: Entry[29] Predecessor: []
NodeName: MethodAccess[59] Predecessor: [ParameterDeclaration[52]]
NodeName: ParameterDeclaration[52] Predecessor: [UncheckedExceptions[46]]
NodeName: UncheckedExceptions[46] Predecessor: [MethodAccess[45]]
NodeName: UncheckedExceptions[60] Predecessor: [MethodAccess[59]]
NodeName: VariableDeclarator[89] Predecessor: [UncheckedExceptions[83], VariableDeclarator[79]]
NodeName: VariableDeclarator[79] Predecessor: [ParameterDeclaration[68]]
NodeName: ParameterDeclaration[68] Predecessor: [UncheckedExceptions[46], UncheckedExceptions[60]]
NodeName: Exit[102] Predecessor: [Entry[93]]
NodeName: Entry[93] Predecessor: []
NodeName: Exit[116] Predecessor: [Entry[115]]
NodeName: Entry[115] Predecessor: []
NodeName: Exit[108] Predecessor: [Entry[107]]
NodeName: Entry[107] Predecessor: []
NodeName: Exit[140] Predecessor: [Entry[139]]
NodeName: Entry[139] Predecessor: []
NodeName: Exit[132] Predecessor: [Entry[131]]
NodeName: Entry[131] Predecessor: []
